examples/src/bin/auto_composite_template.rs

```diff
@@ -0,0 +1,64 @@
+//! # Composite Template Example
+//!
+//! This sample demonstrates how to create a widget using GTK's composite templates.
+
+use gio::prelude::*;
+use glib::subclass::prelude::*;
+use glib::{glib_object_subclass, glib_wrapper};
+use gtk::gtk4_macros::*;
+use gtk::prelude::*;
+
+use gtk::subclass::prelude::*;
+use gtk::subclass::widget::*;
+
+mod imp {
+    use super::*;
+
+    /// The private struct, which will hold the automatically generated widgets
+    #[template(file = "examples/src/bin/composite_template.ui")]
```

Comment:
Would this cause rebuilding if the .ui file changes?

Reply:
I tested it a few times and it does seem to rerun the macro every time the .ui file changes. I think it's actually because it generates include_bytes! somewhere in the output and include_bytes! seems to trigger a rebuild if the file changes.

Reply:
Hmm - include_bytes would mean this doesn't work with gresources - perhaps there needs to be `file` and `resource` arguments separately